Automattic/wp-calypso
# Staging sites: Disable continue button for automated transfers for staging site

Related to https://github.com/Automattic/dotcom-forge/issues/4798

## Proposed Changes

The eligibility endpoint for automated transfer eligibility now returns an error when a site is a non-atomic staging site. This should normally not happen but in very rare circumstances it could occur. In this PR i added handling for this error case.

* Add a new statusMapping for eligibility errors
* Ad a new constant for elibilityHolds
* Add a new message for the hold list to be displayed when the staging transfer eligibility error is returned from the API.

## Testing Instructions

Test this while wpcom is sandboxed and has D146257-code applied. Check out the diff for testing instructions.

When clicking on 'Activate' to enable hosting features, the next page should have the 'Continue' button disabled and display a message explaining why the site can't be transferred.

This is how it should look:
<img width="759" alt="CleanShot 2567-04-23 at 17 15 40@2x" src="https://github.com/Automattic/wp-calypso/assets/10244734/6023d500-30a7-4433-8a12-95dad7c200e6">



*

## Pre-merge Checklist
